Data/ByteString.hs

```diff
@@ -1527,6 +1531,17 @@ stripSuffix bs1@(BS _ l1) bs2@(BS _ l2)
 isInfixOf :: ByteString -> ByteString -> Bool
 isInfixOf p s = null p || not (null $ snd $ breakSubstring p s)
 
+-- | /O(n)/ Check whether a 'ByteString' represents valid UTF-8.
+--
+-- @since 0.11.2.0
+isValidUtf8 :: ByteString -> Bool
```

Comment:
```suggestion
isValidUtf8 :: ByteString -> Bool
isValidUtf8 (BS _ 0) = True
```

Otherwise an empty string causes a segfault in `clang`.

Reply:
Yeah, I even know why. I can fix it in the underlying C - it's curious that GCC 'does the right thing' here, but Clang is stricter and segfaults.